ironfish/src/network/utils/serializers.ts

```diff
@@ -19,55 +15,56 @@ const BLOCK_TRANSACTIONS_LENGTH_BYTES = 2
 
 export function writeBlockHeader(
   bw: bufio.StaticWriter | bufio.BufferWriter,
-  header: SerializedBlockHeader,
+  header: BlockHeader,
 ): bufio.StaticWriter | bufio.BufferWriter {
   bw.writeU32(header.sequence)
   bw.writeHash(header.previousBlockHash)
   bw.writeHash(header.noteCommitment.commitment)
   bw.writeU32(header.noteCommitment.size)
   bw.writeHash(header.nullifierCommitment.commitment)
   bw.writeU32(header.nullifierCommitment.size)
-  bw.writeBytes(BigIntUtils.toBytesLE(BigInt(header.target), 32))
-  bw.writeBytes(BigIntUtils.toBytesLE(BigInt(header.randomness), 8))
-  bw.writeU64(header.timestamp)
+  bw.writeBytes(BigIntUtils.toBytesLE(header.target.targetValue, 32))
+  bw.writeBytes(BigIntUtils.toBytesLE(header.randomness, 8))
+  bw.writeU64(header.timestamp.getTime())
 
-  Assert.isTrue(BigInt(header.minersFee) <= 0)
-  bw.writeBytes(BigIntUtils.toBytesLE(-BigInt(header.minersFee), 8))
+  Assert.isTrue(header.minersFee <= 0)
+  bw.writeBytes(BigIntUtils.toBytesLE(-header.minersFee, 8))
 
-  bw.writeBytes(GraffitiSerdeInstance.deserialize(header.graffiti))
+  Assert.isTrue(header.graffiti.byteLength === 32)
+  bw.writeBytes(header.graffiti)
   return bw
 }
 
-export function readBlockHeader(reader: bufio.BufferReader): SerializedBlockHeader {
+export function readBlockHeader(reader: bufio.BufferReader): BlockHeader {
   const sequence = reader.readU32()
-  const previousBlockHash = reader.readHash('hex')
+  const previousBlockHash = reader.readHash()
   const noteCommitment = reader.readHash()
   const noteCommitmentSize = reader.readU32()
-  const nullifierCommitment = reader.readHash('hex')
+  const nullifierCommitment = reader.readHash()
   const nullifierCommitmentSize = reader.readU32()
-  const target = BigIntUtils.fromBytesLE(reader.readBytes(32)).toString()
-  const randomness = BigIntUtils.fromBytesLE(reader.readBytes(8)).toString()
+  const target = BigIntUtils.fromBytesLE(reader.readBytes(32))
+  const randomness = BigIntUtils.fromBytesLE(reader.readBytes(8))
   const timestamp = reader.readU64()
-  const minersFee = (-BigIntUtils.fromBytesLE(reader.readBytes(8))).toString()
-  const graffiti = GraffitiSerdeInstance.serialize(reader.readBytes(32))
+  const minersFee = -BigIntUtils.fromBytesLE(reader.readBytes(8))
+  const graffiti = reader.readBytes(32)
 
-  return {
+  return new BlockHeader(
     sequence,
-    previousBlockHash,
-    noteCommitment: {
+    Buffer.from(previousBlockHash),
```

Comment:
`previousBlockHash` should be a Buffer here, so we can drop the `Buffer.from`.

Reply:
Changed